diff -urN cvs-nserver-1.11.1.52.org/src/rcs.c cvs-nserver-1.11.1.52/src/rcs.c
--- cvs-nserver-1.11.1.52.org/src/rcs.c	2004-05-23 02:51:33.247861160 +0200
+++ cvs-nserver-1.11.1.52/src/rcs.c	2004-05-23 02:52:29.358331072 +0200
@@ -4230,7 +4230,7 @@
 	    /* If the size of `devtype' changes, fix the sscanf call also */
 	    char devtype[16];
 
-	    if (sscanf (info->data, "%16s %lu",
+	    if (sscanf (info->data, "%15s %lu",
 			devtype, &devnum_long) < 2)
 		error (1, 0, "%s:%s has bad `special' newphrase %s",
 		       workfile, vers->version, info->data);
