--- cvs-nserver-1.11.1.2/doc/cvsclient.texi~	Sat May 19 14:21:42 2001
+++ cvs-nserver-1.11.1.2/doc/cvsclient.texi	Mon Jul 23 04:20:22 2001
@@ -3,9 +3,9 @@
 @setfilename cvsclient.info
 @include CVSvn.texi
 
-@dircategory Programming
+@dircategory Programming tools:
 @direntry
-* cvsclient: (cvsclient).      The CVS client/server protocol.
+* cvsclient: (cvsclient).		The CVS client/server protocol
 @end direntry
 
 @node Top
--- cvs-nserver-1.11.1.2/doc/cvs.texinfo~	Sat Jun 16 21:14:35 2001
+++ cvs-nserver-1.11.1.2/doc/cvs.texinfo	Mon Jul 23 04:21:55 2001
@@ -49,9 +49,10 @@
 
 @ifinfo
 @format
-START-INFO-DIR-ENTRY
-* CVS: (cvs).          Concurrent Versions System
-END-INFO-DIR-ENTRY
+@dircategory Programming tools:
+@direntry
+* CVS: (cvs).				Concurrent Versions System
+@end direntry
 @end format
 @end ifinfo
 
@@ -87,9 +87,9 @@
 @sp 4
 @comment The title is printed in a large font.
 @center @titlefont{Version Management}
-@sp
+@sp 1
 @center @titlefont{with}
-@sp
+@sp 1
 @center @titlefont{CVS}
 @sp 2
 @center for @sc{cvs} @value{CVSVN}
