diff -urNp -x '*.orig' cvs-nserver-1.11.1.52.org/acl/Makefile.am cvs-nserver-1.11.1.52/acl/Makefile.am
--- cvs-nserver-1.11.1.52.org/acl/Makefile.am	2002-01-19 18:33:05.000000000 +0100
+++ cvs-nserver-1.11.1.52/acl/Makefile.am	2021-10-09 23:42:06.967727032 +0200
@@ -1,5 +1,3 @@
-AUTOMAKE_OPTIONS = ../lib/ansi2knr
-
 INCLUDES = -I$(top_srcdir)/lib
 
 TESTS = check_stringbuf check_user_acl \
diff -urNp -x '*.orig' cvs-nserver-1.11.1.52.org/configure.in cvs-nserver-1.11.1.52/configure.in
--- cvs-nserver-1.11.1.52.org/configure.in	2002-02-10 00:26:03.000000000 +0100
+++ cvs-nserver-1.11.1.52/configure.in	2021-10-09 23:42:06.967727032 +0200
@@ -10,7 +10,6 @@ AM_CONFIG_HEADER(config.h src/options.h)
 # initialize test framework
 AC_CONFIG_TESTDIR(tests)
 AC_CONFIG_FILES([tests/Makefile \
-                 tests/atconfig \
                  tests/package.m4 \
                  tests/cvs:tests/wrapsh.in],
                 [chmod +x tests/cvs])
@@ -22,10 +21,7 @@ AC_PATH_PROG(EXPR, expr)
 # their path, the installer can configure with a path that has GNU m4
 # first and get that path embedded in the installed autoconf and
 # autoheader scripts.
-AC_PROG_GNU_M4
-if test x"$ac_cv_prog_gnu_m4" != xyes; then
-  AC_MSG_ERROR([GNU m4 1.4 is required])
-fi
+AC_CHECK_PROG(M4,m4)
 # This is needed because Automake does not seem to realize there is
 # a AC-SUBST inside AC-PROG-GNU-M4.  Grmph!
 AC_SUBST(M4)
@@ -99,8 +95,6 @@ CFLAGS="$CFLAGS -D_SYSV3"
 LIBS="-lcrypt $LIBS"
 fi
 
-AM_C_PROTOTYPES
-
 AM_MAINTAINER_MODE
 
 AC_HEADER_DIRENT
@@ -201,7 +195,7 @@ AC_SYS_LONG_FILE_NAMES
 
 AC_FUNC_FNMATCH
 if test "$ac_cv_func_fnmatch_works" = no; then
-  LIBOBJS="$LIBOBJS fnmatch.o"
+  AC_LIBOBJ([fnmatch])
 fi
 
 # Try to find connect and gethostbyname.
@@ -315,7 +309,7 @@ if test "$encryption" = "true"; then
   AC_DEFINE(ENCRYPTION)
 fi
 
-AC_CHECK_FUNC(gethostname, :, LIBOBJS="$LIBOBJS hostname.o")
+AC_CHECK_FUNC(gethostname, :, [AC_LIBOBJ([hostname])])
 
 #
 # use --with-openssl to turn on support for OpenSSL
@@ -428,7 +422,7 @@ AC_CACHE_CHECK(for cygwin32, ccvs_cv_sys
 [AC_TRY_COMPILE([], [return __CYGWIN32__;],
 ccvs_cv_sys_cygwin32=yes, ccvs_cv_sys_cygwin32=no)])
 if test $ccvs_cv_sys_cygwin32 = yes; then
-  LIBOBJS="$LIBOBJS fncase.o"
+  AC_LIBOBJ([fncase])
   LIBS="$LIBS -ladvapi32"
 
   dnl On Windows you can only change file times if you can write to
@@ -457,7 +451,6 @@ AC_CONFIG_FILES([Makefile \
 	  lib/Makefile \
 	  man/Makefile \
 	  os2/Makefile \
-	  scripts/Makefile \
 	  src/Makefile \
           src/tests/Makefile \
 	  tools/Makefile \
diff -urNp -x '*.orig' cvs-nserver-1.11.1.52.org/doc/Makefile.am cvs-nserver-1.11.1.52/doc/Makefile.am
--- cvs-nserver-1.11.1.52.org/doc/Makefile.am	2001-10-13 19:47:03.000000000 +0200
+++ cvs-nserver-1.11.1.52/doc/Makefile.am	2021-10-09 23:42:06.967727032 +0200
@@ -14,7 +14,7 @@
 
 #DISTFILES = \
 #	.cvsignore \
-#	RCSFILES \
+#	RCSFILES
 
 # AUTOMAKE_OPTIONS = no-texinfo.tex
 
diff -urNp -x '*.orig' cvs-nserver-1.11.1.52.org/src/Makefile.am cvs-nserver-1.11.1.52/src/Makefile.am
--- cvs-nserver-1.11.1.52.org/src/Makefile.am	2021-10-09 23:42:06.901060450 +0200
+++ cvs-nserver-1.11.1.52/src/Makefile.am	2021-10-09 23:42:06.967727032 +0200
@@ -12,8 +12,6 @@
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 # GNU General Public License for more details.
 
-AUTOMAKE_OPTIONS = ../lib/ansi2knr
-
 SUBDIRS = . tests
 
 # $(includeopt) is CVS specific and set by configure
