diff -urN cvs-nserver-1.11.1.4.org/Makefile.am cvs-nserver-1.11.1.4/Makefile.am
--- cvs-nserver-1.11.1.4.org/Makefile.am	Sun May 19 11:15:31 2002
+++ cvs-nserver-1.11.1.4/Makefile.am	Sun May 19 11:16:39 2002
@@ -17,7 +17,7 @@
 ## Subdirectories to run make in for the primary targets.
 # Unix source subdirs, where we'll want to run lint and etags:
 # This is a legacy variable from b4 Automake
-USOURCE_SUBDIRS = lib zlib diff src
+USOURCE_SUBDIRS = lib diff src
 # All other subdirs:
 SUBDIRS = $(USOURCE_SUBDIRS) man doc contrib tools \
 	windows-NT os2 emx vms
diff -urN cvs-nserver-1.11.1.4.org/configure.in cvs-nserver-1.11.1.4/configure.in
--- cvs-nserver-1.11.1.4.org/configure.in	Sun May 19 11:15:31 2002
+++ cvs-nserver-1.11.1.4/configure.in	Sun May 19 11:16:23 2002
@@ -375,6 +375,8 @@
   AC_DEFINE(HAVE_SETMODE)
 fi
 
+AC_CHECK_LIB(z, zlibVersion)
+
 test -f src/options.h && (
   AC_MSG_WARN(saving ./src/options.h in ./src/options.h-SAVED)
   AC_MSG_WARN(You may wish to check that local options have not been lost.)
@@ -405,8 +407,7 @@
 	  tools/Makefile \
 	  vms/Makefile \
 	  windows-NT/Makefile \
-	  windows-NT/SCC/Makefile \
-	  zlib/Makefile],
+	  windows-NT/SCC/Makefile],
 	  [chmod -f +x \
 	  contrib/clmerge \
 	  contrib/cln_hist \
diff -urN cvs-nserver-1.11.1.4.org/src/Makefile.am cvs-nserver-1.11.1.4/src/Makefile.am
--- cvs-nserver-1.11.1.4.org/src/Makefile.am	Sun May 19 11:15:31 2002
+++ cvs-nserver-1.11.1.4/src/Makefile.am	Sun May 19 11:17:19 2002
@@ -20,7 +20,7 @@
 # try and remove naming conflicts and fix Automake to allow particular includes
 # to be attached only to particular object files.  Short term fix is either or.
 ##INCLUDES = -I. -I.. -I$(srcdir) -I$(top_srcdir)/lib
-INCLUDES = -I$(top_srcdir)/lib -I$(top_srcdir)/diff -I$(top_srcdir)/zlib $(includeopt)
+INCLUDES = -I$(top_srcdir)/lib -I$(top_srcdir)/diff $(includeopt)
 
 bin_PROGRAMS = cvs cvs-nserver cvs-pserver cvschkpw
 bin_SCRIPTS = cvsbug
@@ -114,7 +114,6 @@
 cvs_LDADD = \
 	../diff/libdiff.a \
 	../lib/libcvs.a \
-	../zlib/libz.a \
 	version.o
 cvs_EXTRA_DIST = version.c
 
